root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs

```diff
@@ -99,126 +109,223 @@ private void InitializeComponent()
+        /// calling main thread
         /// </summary>
-        protected override void OnStart(string[] args)
+        public void MainThreadInvoke()
         {
-            StartByThreadPool(AsyncProcessingServiceParameterValue.AsyncStatus.Processing);
+            _asyncParameterValue = new AsyncProcessingServiceParameterValue("AsyncProcessingService", "Select", "Select", "SQL",
+                                                                            new MyUserInfo("AsyncProcessingService", "AsyncProcessingService"));
+
+            AsyncProcessingService.LayerB myBusiness = new AsyncProcessingService.LayerB();
+            AsyncProcessingServiceReturnValue asyncReturnValue = (AsyncProcessingServiceReturnValue)myBusiness.DoBusinessLogic(
+                                                                 (BaseParameterValue)_asyncParameterValue, DbEnum.IsolationLevelEnum.ReadCommitted);
+
+            _asyncParameterValue = (AsyncProcessingServiceParameterValue)asyncReturnValue.Obj;
+
+            //Sets MaxThreads in ThreadPool.
+            bool SetMaxThreadsResult = ThreadPool.SetMaxThreads(10, 10);
+
+            //Main thread assigned task to worker thread in threadpool
+            ThreadStart threadStart = new ThreadStart(() => StartByThreadPool(_asyncParameterValue));
+            _workerThread = new Thread(threadStart);
+            _workerThread.Start();
+
+            //Update in mainthread
+            _asyncParameterValue = new AsyncProcessingServiceParameterValue("AsyncProcessingService", "Update", "Update", "SQL",
+                                                                            new MyUserInfo("AsyncProcessingService", "AsyncProcessingService"));
+
+            DbEnum.IsolationLevelEnum iso = DbEnum.IsolationLevelEnum.DefaultTransaction;
+
+            asyncReturnValue = (AsyncProcessingServiceReturnValue)myBusiness.DoBusinessLogic((AsyncProcessingServiceParameterValue)_asyncParameterValue, iso);
         }
 
+        #endregion
+
+        #region StartByThreadPool
+
         /// <summary>
         /// Calls callback method using Threadpool worker item.
         /// </summary>
-        /// <param name="check"></param>
+        /// <param name="_asyncParameterValue"></param>
         /// <returns></returns>
-        public bool StartByThreadPool(AsyncProcessingServiceParameterValue.AsyncStatus check)
+        public bool StartByThreadPool(AsyncProcessingServiceParameterValue _asyncParameterValue)
+        {
+            ThreadPool.QueueUserWorkItem(new WaitCallback(InvokeController), (object)_asyncParameterValue.ProcessName);
+            return true;
+        }
+
+        #endregion
+
+        #region InvokeController
+
+        /// <summary>
+        /// Updates using LayerB through callcontroller class
+        /// </summary>
+        /// <param name="_asyncParameterValue"></param>
+        private void InvokeController(object _asyncParameterValue)
```

Comment:
There is a problem with the argument name.
You need naming to argument as "processName".